### DISCUSSION TITLE
Work directly with Files instead of path of files

### DISCUSSION THREAD

**Discussion by ilyasbendev** [openai/whisper#1620] [Q&A] [answered]
Hello everyone,
I'm working with whisper actually, my goal is to transcribe an audio file.
It's ok when i enter the path of file as an input but its not working when i'm dealing with files directly.
I did the instructions of this discussion : https://github.com/openai/whisper/discussions/216?sort=new  , but its doesn't work.
When i run the code it take me a huge time without any results (i don't have any errors or exceptions when running).

Is there any idea to resolve this issue ?

Thanks in advance.

**Comment by threeal** [comment]
Try to use [librosa](https://pypi.org/project/librosa/) to load audio data from file buffer and pass that to `model.transcribe`:

```py
import io
import librosa
import whisper

model = whisper.load_model("tiny")

with open("path/to/file", "rb") as file:
    buf = io.BytesIO(file.read())
    data, sr = librosa.load(buf)
    
    # Resampling might be required here
    if sr != 16000:
        data = librosa.resample(data, orig_sr=sr, target_sr=16000)
    
    model.transcribe(data)
```

Or if you already have a file buffer, try to directly pass that to `librosa.load`

**Reply by ilyasbendev** [reply, marked as answer]
Thank you @threeal for your response :
I got this error : 

**can only concatenate str (not "dict") to str** 
when calling the transcribe() method 

This is the code that i wrote :
<pre>
audio_file = request.files['media'] // This is how i got the file when calling my endpoint (i'm using flask)
buf = io.BytesIO(audio_file.read())
data, sr = librosa.load(buf)

if sr != 16000:
   print('data is different than 16000')
   data = librosa.resample(data, orig_sr=sr, target_sr=16000)
   
model = whisper.load_model("tiny")
model.transcribe(data,fp16=False)
</pre>


I think that the problem is in this line of code :
<pre>
buf = io.BytesIO(audio_file.read())
</pre>
Many thanks in advance.

**Reply by threeal** [reply]
can you show me the full stack trace?

probably you just need to pass the `audio_file` without calling `read()` like this:
```py
buf = io.BytesIO(audio_file)
```

or pass it to `librosa.load` directly:
```py
librosa.load(audio_file)
```

i'm not sure what is the type of `audio_file` on your side, so it's hard to figured out.

**Reply by ilyasbendev** [reply]
@threeal this is not working also
This is how i got the audio_file : (i'm working with flask, i got the file directly from the request)
<pre>
def post(self):
        if 'media' not in request.files:
            return {'message': 'No media file provided'}, 400

        audio_file = request.files['media']
 </pre>

**Reply by ilyasbendev** [reply]
@threeal ,
It was a wrong manip for me, your first suggestion was good.
Thanks a lot.
